public/mobile-app/src/lib/api-holidays.test.ts

```diff
@@ -0,0 +1,130 @@
+import { afterEach, describe, test, expect, vi } from 'vitest'
+import '@testing-library/jest-dom/vitest'
+import {
+  dayName,
+  monthName,
+  holidayPeriod,
+  retrieveHolidays,
+} from '$lib/api-holidays.ts'
+
+describe('/api-particulier.ts', () => {
+  describe('dayName', () => {
+    test('should return short day name', async () => {
+      // Given
+      const date = new Date('2025-11-11')
+
+      // When
+      const name = dayName(date)
+
+      // Then
+      expect(name).equal('mar')
+    })
+  })
+  describe('monthName', () => {
+    test('should return long month name', async () => {
+      // Given
+      const date = new Date('2025-11-11')
+
+      // When
+      const name = monthName(date)
+
+      // Then
+      expect(name).equal('Novembre')
+    })
+  })
+  describe('holidayPeriod', () => {
+    test('should not mention start date year', async () => {
+      // Given
+      const startDate = new Date('2025-10-15')
+      const endtDate = new Date('2025-11-15')
+
+      // When
+      const period = holidayPeriod(startDate, endtDate)
+
+      // Then
+      expect(period).equal('Du 15 octobre au 15 novembre 2025')
+    })
+    test('should not mention start date year and month', async () => {
+      // Given
+      const startDate = new Date('2025-10-15')
+      const endtDate = new Date('2025-10-20')
+
+      // When
+      const period = holidayPeriod(startDate, endtDate)
+
+      // Then
+      expect(period).equal('Du 15 au 20 octobre 2025')
+    })
+    test('should mention start date year and month', async () => {
+      // Given
+      const startDate = new Date('2025-12-20')
+      const endtDate = new Date('2026-01-02')
+
+      // When
+      const period = holidayPeriod(startDate, endtDate)
+
+      // Then
+      expect(period).equal('Du 20 décembre 2025 au 2 janvier 2026')
+    })
+  })
+  describe('retrieveHolidays', () => {
+    afterEach(() => {
+      window.localStorage.clear()
+    })
```

Comment:
Nit: is the code using localstorage at all? Is this useful?

Reply:
No, it is not usefull, thanks :)